### ISSUE TITLE
Bbox type incompatible with @types/geojson

### ISSUE THREAD

**Issue by graue** [reilem/zod-geojson#4]
Hi, thanks for making this!

I'm having trouble getting a polygon parsed with zod-geojson to be accepted as a `GeoJSON.Polygon` as defined in the `@types/geojson` package. It looks like the problem might be that `@types/geojson` [defines a bbox as a tuple of 4 or 6 numbers](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/40bfaf269fbb216b79acd2aa25d30d2123d01bba/types/geojson/index.d.ts#L22):

```.ts
export type BBox = [number, number, number, number] | [number, number, number, number, number, number];
```

while this lib [defines it as an array of 4 or more elements](https://github.com/reilem/zod-geojson/blob/67911dc7d5daaa5a4195a9d91073515d31eba8e0/src/bbox.ts#L3):

```.ts
export type GeoJSONBbox = [number, number, number, number, ...number[]];
```

Is there any reason for that flexibility as opposed to expecting 4 or 6 elements?

**Comment by reilem** [comment]
My implementation follows the RFC:

https://datatracker.ietf.org/doc/html/rfc7946#section-5

> A GeoJSON object MAY have a member named "bbox" to include
   information on the coordinate range for its Geometries, Features, or
   FeatureCollections.  The value of the bbox member MUST be an array of
   length 2*n where n is the number of dimensions represented in the
   contained geometries, with all axes of the most southwesterly point
   followed by all axes of the more northeasterly point.  The axes order
   of a bbox follows the axes order of geometries.

So a bbox can have: 4, 6, 8, 10, ... elements depending on the dimension of the geojson. So it does seem like `@types/geojson`is incorrect in this regard. If you need a specific dimension then you can try using `GeoJSON2DPolygonSchema` or `GeoJSON3DPolygonSchema` 🙂

**Comment by graue** [comment]
And the RFC gives examples of 2- and 3-dimensional data. In what science-fiction universe does geospatial data include a fourth dimension?

In any event, I get the same error using `GeoJSON2DPolygonSchema`; it doesn't seem to refine the output bbox type for some reason.

**Comment by reilem** [comment]
Hi @graue, you're right it doesn't make much sense, but I tried to implement this to be as compatible with other implementations as possible. The RFC states the following:

> Implementations SHOULD NOT extend positions beyond three elements
   because the semantics of extra elements are unspecified and
   ambiguous. 

Since it phrases this as "should not" and not "must not" I did not want to make this library unusable by anyone who has defined some fourth dimension for their internal geojson structure. E.g:

> Historically, some implementations have used a fourth
   element to carry a linear referencing measure (sometimes denoted as
   "M") or a numerical timestamp, but in most situations a parser will
   not be able to properly interpret these values.  The interpretation
   and meaning of additional elements is beyond the scope of this
   specification, and additional elements MAY be ignored by parsers.

I guess since it states that I may ignore additional elements. I may simplify the typings to not force myself to have to support this. I will let you know when it's fixed. Thanks for bringing this to my attention!

**Comment by graue** [comment]
That makes sense! Thanks. It would be helpful for me to have the parsed types compatible with `@types/geojson`.

**Comment by reilem** [comment]
I just released version 0.0.3, there are still some issues with the @types/geojson package which I cannot solve with zod-geojson without compromising the correctness. You can find more details here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/71066

**Comment by graue** [comment]
0.0.3 does solve my issue! Thank you :pray: Appreciate your careful attention to correctness.
